internal/entities/subscription.go

```diff
@@ -10,7 +11,21 @@ type CurrencyDispatchDetails struct {
 }
 
 type Dispatch[T any] struct {
```

Comment:
What value you see in generic usage here?

Reply:
At the moment it is extra, but in perspective it dispatch service could sent emails not just about currency rates (e.g. payment receipt)

Reply:
It's ok to left it here.
Just remember one thing "if you can to not do something extra, better not to do it". ( but we must think about extensibility of our code")